components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go

```diff
@@ -0,0 +1,109 @@
+package sshuseraccesschecktrigger
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/configutils"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/httputils"
+	"github.com/chef/automate/lib/logger"
+)
+
+type SshUserAccessCheck struct {
+	host string
+	port string
+	log  logger.Logger
+}
+
+func NewSshUserAccessCheck(log logger.Logger, port string) *SshUserAccessCheck {
+	return &SshUserAccessCheck{
+		log:  log,
+		host: constants.LOCAL_HOST_URL,
+		port: port,
+	}
+}
+
+func (ss *SshUserAccessCheck) Run(config models.Config) []models.CheckTriggerResponse {
+	ss.log.Info("Performing SSH user access check from batch check ")
+	count := config.Hardware.AutomateNodeCount + config.Hardware.ChefInfraServerNodeCount +
+		config.Hardware.PostgresqlNodeCount + config.Hardware.OpenSearchNodeCount
+
+	outputCh := make(chan models.CheckTriggerResponse, count)
+
+	var finalResult []models.CheckTriggerResponse
+
+	for _, ip := range config.Hardware.AutomateNodeIps {
```

Comment:
Can we have one loop that can have dynamic ip, nodeType Values and get rid of 4 loops ?